### PR TITLE
Maya: Enhancement on reset frame range based on fps

### PR DESCRIPTION
**Pull request by moonyuet**
## Changelog Description
- Support for fps options in terms of maya settings(such as 2, 3, 4, 5, 6, 8, 10, 12, 16 fps)
- Support to set fps value based on the task

## Additional info
n/a

## Testing notes:
1. Go to ayon server -> Project Setting -> project and select your project
2. Set the fps value based for your task
image
3. Launch Maya
4. Access AYON menubar -> Set frame range
5. Should be set the fps value according to the project setting.

### CODE REVIEW

File: client/ayon_core/hosts/maya/api/lib.py [hunk 1]
```diff
@@ -2520,7 +2520,17 @@ def set_scene_fps(fps, update=True):
     """
 
     fps_mapping = {
+        # 2, 3, 4, 5, 6, 8, 10, 12, 16
```
Comment (tokejepsen) [inline client/ayon_core/hosts/maya/api/lib.py:2523]:
```suggestion
```
Dont think this comment adds anything.

**Review by tokejepsen — CHANGES_REQUESTED** [review]
Tested successfully in Maya 2023. Just that small code cosmetics suggestion.

**Review by tokejepsen — APPROVED** [review]
Ready to merge.

File: client/ayon_core/hosts/maya/api/lib.py [hunk 2]
```diff
@@ -2612,21 +2621,24 @@ def get_fps_for_current_context():
...
+        fps = folder_entity.get("attrib", {}).get("fps")
         if not fps:
-            fps = 25
+            project_entity = ayon_api.get_project(
+                project_name, fields=["attrib.fps"]
+            ) or {}
+            fps = project_entity.get("attrib", {}).get("fps")
+
+            if not fps:
+                fps = 25
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply client/ayon_core/hosts/maya/api/lib.py:2641]: A little bit late, but yes it is the case. There are more places that should be cleaned up over time, usually frame start/end and fps.

**Pull request merged**